src/Microsoft.TemplateEngine.Cli/CommandParsing/CommandLineHelpViewerExtensions.cs

```diff
@@ -0,0 +1,111 @@
+using Microsoft.DotNet.Cli.CommandLine;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Cli.CommandParsing
+{
+    internal static class CommandLineHelpViewerExtensions
+    {
+        private const int ColumnGutterWidth = 3;
+        private const string TemplateArgument = "TEMPLATE";
+
+        internal static string HelpView(this Command newCommand)
+        {
+            StringBuilder helpView = new StringBuilder();
+            WriteSynopsys(helpView, newCommand);
+            WriteArgumentsSection(helpView);
+            WriteOptionsSection(helpView, newCommand);
+            WriteExamples(helpView, newCommand);
+            return helpView.ToString();
+        }
+
+        private static void WriteSynopsys(StringBuilder helpView, Command command)
+        {
+            helpView.AppendLine(string.Format(LocalizableStrings.NewCommandUsage, command.Name, TemplateArgument));
+            helpView.AppendLine();
+        }
+
+        private static void WriteArgumentsSection(StringBuilder helpView)
+        {
+            helpView.AppendLine(LocalizableStrings.ArgumentsSectionTitle);
+            WriteColumnizedSummary($"  <{TemplateArgument}>", string.Format(LocalizableStrings.ArgumentDescriptionTemplate, TemplateArgument), 20, helpView);
+            helpView.AppendLine();
+        }
+
+        private static void WriteOptionsSection(StringBuilder helpView, Command command)
+        {
+            Option[] options = command
+                .DefinedOptions
+                .Where(o => !o.IsHidden())
+                .ToArray();
+
+            if (!options.Any())
+            {
+                return;
+            }
+
+            helpView.AppendLine(LocalizableStrings.OptionsSectionTitle);
+            WriteOptionsList(options, helpView);
+            helpView.AppendLine();
+        }
+
+
+        private static void WriteOptionsList(Option[] options, StringBuilder helpView)
+        {
+            Dictionary<Option,string> leftColumnTextFor = options.ToDictionary(o => o, LeftColumnText);
+
+            int leftColumnWidth = leftColumnTextFor
+                                      .Values
+                                      .Select(s => s.Length)
+                                      .OrderBy(length => length)
```

Comment:
We can just use `Max` extension method from `Linq` to get the string with the largest length. Sorting can be avoided in that case.
Or am I just misunderstanding the intent here?